src/iop/colorbalancergb.c

```diff
@@ -808,7 +808,7 @@ void process(struct dt_iop_module_t *self,
       scaled_RGB[c] = abs_RGB[c] /d->white_fulcrum;
     dt_vector_powf(scaled_RGB, midtones, RGB);
     for_each_channel(c)
-      RGB[c] = RGB[c] * sign[c] * d->white_fulcrum;
+      RGB[c] = MAX(0.0f, RGB[c] * sign[c] * d->white_fulcrum);
```

Comment:
I have a doubt though, what about the sign? As noted for the OpenCL case we really do want to keep the sign:

`// midtones : power with sign preservation`

The `fmaxf()` does break this.